sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs

```diff
@@ -297,110 +282,52 @@ public virtual async Task<Response<UploadLogsResult>> UploadAsync<T>(string rule
                         // Clear any completed blocks from the task list
                         for (int i = 0; i < runningTasks.Count; i++)
                         {
-                            Task<BatchUpload> runningTask = runningTasks[i];
+                            Task<Response> runningTask = runningTasks[i];
                             if (!runningTask.IsCompleted)
                             {
                                 continue;
                             }
-
-                            // Before removing runningTask - check if it has any errors and add to error list
-                            if (runningTask.Result.Error != null)
-                                errors.Add(runningTask.Result.Error);
-
                             // Remove completed task from task list
                             runningTasks.RemoveAt(i);
                             i--;
                         }
                     }
                 }
-
-                // Wait for all the remaining blocks to finish uploading
-                await Task.WhenAll(runningTasks).ConfigureAwait(false);
-
-                // Process all errors after tasks are done to determine status
-                // Will run on a single thread
-                foreach (Task<BatchUpload> task in runningTasks)
+                catch (Exception ex)
                 {
-                    // Go through errors from each task and if we have an error, add error from response into errors list which represents the errors from all the batches
-                    if (task.Result.Error != null)
-                        errors.Add(task.Result.Error);
+                    // If we have an error, add error from response into exceptions list which represents the errors from all the batches
+                    exceptions.Add(ex);
                 }
             }
-            catch (Exception ex)
-            {
-                scope.Failed(ex);
-                throw;
-            }
 
-            // Calculate the status using the helper method Status
-            UploadLogsResult finalResult = new UploadLogsResult(errors, GetStatus(logs, errors));
-            return Response.FromValue(finalResult, response);
-        }
+            // Wait for all the remaining blocks to finish uploading
+            await Task.WhenAll(runningTasks).ConfigureAwait(false);
```

Comment:
Does this need to happen inside your try/catch block for any exceptions that might get raised?

Reply:
If we await all tasks using WhenAll, we can use this [workaround](https://github.com/dotnet/runtime/issues/31494#issuecomment-553895267) to make sure an AggregateException is thrown for multiple exceptions.